Split out list service

alex-deaconu opened the pull request:
Using it for presentations and products lists

[stage-2]

alex-deaconu commented:
@ezequielc @rodrigopavezi can you please take a look at this PR?

Basically, I've created a service that facilitates Scrolling List functionality for any controller. It will create the search object with default values (or receive one from the controller), and provide all the list functions that the controller needs: load, search and sort.

Pretty much prevents duplicate code for lists.

Please let me know if you have any comments or suggestions on how to improve this.

Thanks!

ezequielc commented on js/services/svc-scrolling-list-service.js:
```diff
@@ -0,0 +1,63 @@
+              })
+              .then(null, function (e) {
+                factory.error =
+                  'Failed to load presentations. Please try again later.';
```
@alex-deaconu This message should not refer to presentations, right?

alex-deaconu replied:
@ezequielc good catch, will fix. Thanks.

ezequielc commented on js/services/svc-scrolling-list-service.js:
```diff
@@ -0,0 +1,63 @@
+
+        factory.search = search ? search : {};
+        _.defaults(factory.search, {
+          sortBy: 'changeDate',
```
@alex-deaconu what happens if the item does not have changeDate property?

alex-deaconu replied:
@ezequielc good question. From an API standpoint, nothing should happen, it should just ignore the sort parameter. Do you have any suggestions on how I could deal with this case?

However, you bring up a good point. Generally lists are either sorted by `changeDate` or `name`. I just assumed `changeDate` is the more common of the two, so I added it here as default. Do you think I should default to `name`? 

Of course, the value can be overwritten by providing a `search` object.

Thanks!

ezequielc replied:
@alex-deaconu Not sure also. One idea would be to leave `search` empty and get the default sort from the endpoint API. 
But as you said, the value can be overwritten, so it is not a big problem. Thanks!

ezequielc commented:
@alex-deaconu Just two minor comments. Looks good. :+1:

rodrigopavezi commented:
@alex-deaconu LG :+1:

alex-deaconu commented:
@ezequielc unfortunately if the sort is missing or using a wrong column name, the list returns in a very random order. I changed the sort by default to `name`, this column should be there most of the time. Also fixed the typo. Thanks.
